engine/source/public/core/asset.hpp

```diff
@@ -0,0 +1,71 @@
+﻿#pragma once
+
+#include <string>
+
+namespace lumina
+{
+    enum class AssetType
+    {
+        Texture, // Including Samplers or not can be discussed
+        Shader,
+        Model, // Collections of meshes
+        Mesh,
+        Material,
+        None
+    };
+
+    class Asset
+    {
+        friend class AssetManager;
+
+    private:
+        static size_t nextID;
+
+    protected:
+        size_t id;
+        AssetType type;
+        std::string filePath;
+
+        Asset(const AssetType type) : id(0), type(type) {}
+
+        virtual ~Asset();
+
+        virtual void Reload() {}
+
+        [[nodiscard]] static size_t GenerateID(const std::string& path)
```

Comment:
should `GenerateID` do something with `nextID`? because i would expect those 2 to be related.
for example:
```c++
int id1 = GenerateID(path);
int id2 = NextID();
```

Reply:
ping